client/src/services/gameService/index.ts

```diff
@@ -1,27 +1,50 @@
 import { Socket } from "socket.io-client";
+
 class GameService {
 
     // This function is called from our homePage to tell the server to host a new room and send us the code
-    public async hostRoom(socket: Socket): Promise<string> {
+    public async startRound(socket: Socket, roundId: number): Promise<string[]> {
         return new Promise((rs, rj) => {
-            socket.emit("host_room");
-            socket.on("room_host_success", (message) => rs(message));
-            socket.on("room_host_error", ({ error }) => rj(error));
+            socket.emit("start_round", roundId);
+            socket.on("send_question", (question) => rs(question));
+            socket.on("start_game_error", ({ error }) => rj(error));
         })
     }
 
-    // This function tells the server to join a room with roomId and our username and returns if it worked or not
-    public async joinRoom(socket: Socket, roomId: string, username: string): Promise<boolean> {
+    // This function is called from the player to send their answer
+    public async sendAnswer(socket: Socket, answerId: string): Promise<string> {
         return new Promise((rs, rj) => {
-            socket.emit("join_game", { roomId, username });
-            socket.on("room_join_success", () => rs(true));
-            socket.on("room_join_error", ({ error }) => rj(error));
+            socket.emit("send_answer", answerId);
+
+            // Trying to test slow loading time
+            socket.on("answer_received", (message) => rs(message));
+            socket.on("answer_error", ({ error }) => rj(error));
         })
     }
 
-    // This function allows us to listen for new players joining and sends the list of players in the room if that changes
-    public async onPlayerJoin(socket: Socket, listener: (message: any) => void) {
-        socket.on("on_player_join", (message) => listener(message));
+    // This function allows the players to listen for new questions
+    public async onSendQuestion(socket: Socket, listener: (q: string[]) => void) {
+        socket.on("send_question", (question) => listener(question));
+    }
+
+    // This function allows the host to listen to new answers sent in by players
+    public async onUpdateAnswers(socket: Socket, listener: (message: any) => void) {
```

Comment:
So, for the host to gather all the answers from the players, the host needs to listen to onUpdateAnswers() in useEffect and collect the answers (maybe in a global state)? Does this send the answer with the username so the host can keep track of the score to display at the end of the round, or just the answer alone?